crates/aggchain-proof-builder/src/lib.rs

```diff
@@ -281,68 +290,82 @@ impl<ContractsClient> AggchainProofBuilder<ContractsClient> {
         };
 
         {
-            let retrieved_from_contracts = AggregationProofPublicValues::from(&fep_inputs);
-
-            if request.aggregation_proof_public_values != retrieved_from_contracts {
-                error!(
-                    "Mismatch between the aggregation proof public values - retrieved from the \
-                     contracts: {retrieved_from_contracts:?}, received with the proof: {:?}",
-                    request.aggregation_proof_public_values
-                );
-                return Err(Error::MismatchAggregationProofPublicValues {
-                    expected_by_contract: Box::new(retrieved_from_contracts),
-                    expected_by_verifier: Box::new(request.aggregation_proof_public_values),
-                });
+            if let FepVerification::Proof {
+                ref aggregation_proof_public_values,
+                ..
+            } = request.fep_verification
```

Comment:
else if `FepVerification::Optimistic { signature }` then to set `fep_inputs.signature_optimistic_mode = Some(signature)`

I think this could be plugged in your PR right?